Build failure: audacity

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

commit 710771af3d1c8c3f86a9e5d562616973ed5f3f21
build audacity

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/303188112

### Relevant log output

```console
[31mCMake Error at cmake-proxies/cmake-modules/dependencies/rapidjson.cmake:6 (add_library):
  add_library cannot create ALIAS target "rapidjson::rapidjson" because
  target "rapidjson" is itself an ALIAS.
Call Stack (most recent call first):
  cmake-proxies/cmake-modules/AudacityDependencies.cmake:230 (include)
  CMakeLists.txt:592 (include)

[0m
-- Trying to retrieve GTK version from general;-L/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib;general;-pthread;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_baseu-3.2.so;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_gtk3u_core-3.2.so;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_gtk3u_html-3.2.so;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_gtk3u_qa-3.2.so;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_baseu_xml-3.2.so;general;/nix/store/a4v3pmhzkdz4cn6r8x2mfy9ybwhrygcl-wxwidgets-3.2.8.1/lib/libwx_baseu_net-3.2.so
-- Found PkgConfig: /nix/store/7vrzhnby95407cbbahn3fj3y98qifgby-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.49
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Checking for module 'glib-2.0'
--   Found glib-2.0, version 2.84.3
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Fixing up ZLib mess...
-- ========== Configuring vamp ==========
-- Using 'vamp' local library
-- ========== Configuring lv2 ==========
-- Using 'lv2' local library
-- Checking for module 'x11'
--   Found x11, version 1.8.12
-- Checking for module 'gtk+-2.0'
--   No package 'gtk+-2.0' found
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.49
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-
...
-- Configuring incomplete, errors occurred!
```

### Additional context

[log.txt](https://github.com/user-attachments/files/21452376/log.txt)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.0-rc5, NixOS, 25.11 (Xantusia), 25.11.20250712.b1b18ff`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/kpdma5q42796jk8ky5fx9y8xfih07kgj-source`

### Notify maintainers

@veprbl @wegank

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/au/audacity/package.nix
+++ b/pkgs/by-name/au/audacity/package.nix
@@ -70,6 +70,11 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-kESKpIke9Xi4A55i3mUu1JkDjp8voBJBixiAK8pUkKA=";
   };
 
+  patches = [
+    # Introduced by https://github.com/Tencent/rapidjson/commit/b1c0c2843fcb2aca9ecc650fc035c57ffc13697c#diff-2f1bcf2729ff7c408adb0c2cc2cfa01602bd5646b05b3e4bc7e46b606035d249R21
+    ./rapidjson.patch
+  ];
+
   postPatch = ''
     mkdir src/private
     substituteInPlace scripts/build/macOS/fix_bundle.py \

--- a/pkgs/by-name/au/audacity/rapidjson.patch
+++ b/pkgs/by-name/au/audacity/rapidjson.patch
@@ -1,0 +1,18 @@
+diff --git a/cmake-proxies/cmake-modules/dependencies/rapidjson.cmake b/cmake-proxies/cmake-modules/dependencies/rapidjson.cmake
+index ba95962..66a96a8 100644
+--- a/cmake-proxies/cmake-modules/dependencies/rapidjson.cmake
++++ b/cmake-proxies/cmake-modules/dependencies/rapidjson.cmake
+@@ -2,8 +2,11 @@
+ 
+ if(NOT ${_OPT}use_rapidjson STREQUAL "off")
+    if(NOT TARGET rapidjson::rapidjson)
+-      if(TARGET rapidjson)
+-         add_library( rapidjson::rapidjson ALIAS rapidjson )
++      if(TARGET RapidJSON)
++         if(NOT TARGET rapidjson)
++            add_library(rapidjson ALIAS RapidJSON)
++         endif()
++         add_library( rapidjson::rapidjson ALIAS RapidJSON )
+       else()
+          # At least on Arch RapidJSONConfig.cmake does not define a target at all
+          # so we have to do it ourselves

